wordpress-mobile/WordPress-iOS
# Notifications Control Center: New Interface

### Notifications Settings: Interface Layer
We're adding the interface that will be rendering the new Notification Settings Service, introduced in PR #3917. Below you'll find the Test Scenarios, required to validate this functionality.

--


#### Test Scenario A: Nuked Legacy Push Switch
1. Launch WPiOS and log into a Dotcom Account
2. Tap over `My Sites`
3. Pick any site
4. Tap over Settings

Make sure that the `Push Notifications` Flip Switch Cell is gone.


-- 


#### Test Scenario B: Notification Settings only for Dotcom
1. Launch WPiOS and log into a Self Hosted site
2. Tap over the `Me` Tab
3. Note that there shouldn't be a `Notifications` row.
4. Log into your WordPress.com account

As a result, the `Notifications` row should show up.


--


#### Test Scenario C: Manage Notifications
1. Launch WPiOS and log into a dotcom account
2. Tap over the `Notifications` tab

The top right icon (`Manage`) should be gone.


--


#### Test Scenario D: Load More Settings
1. Launch WPiOS
2. Log into an account that has +3 DotCom blogs
3. Tap over `Me` > `Notifications`
4. Note that only the first three blogs should show up
5. Tap over the `View All...` cell

As a result, all of the blogs that support Notifications should show up. Please, test this scenario in both, iPad + iPhone.


--


#### Test Scenario E: Primary Blog
1. Launch WPiOS and log into a dotcom account
2. Tap over `Me` >  `Notifications`

Verify that the Primary Blog should show up on top.


--


#### Test Scenario F: Updating Settings
1. Launch WPiOS and log into a dotcom account
2. Tap over `Me` >  `Notifications`
3. Open the settings for any of your blogs
4. Change a setting
5. Hit `Back` until you reach the main `Me` tab
6. Open the Notification Settings for the site that you just updated

As a result, the updated settings should be onscreen.
Please, repeat this procedure for `Other Sites` and `WordPress.com` sections as well


--


#### Test Scenario G: Update Error Handling
1. Launch WPiOS and log into a dotcom account
2. Tap over `Me` >  `Notifications`
3. Open the settings for any of your blogs
4. Change a setting
5. Enable your phone's *airplane mode*
6. Hit back

As a result, you should get an alertView indicating that there was an error. This alertView should allow you to `Retry` / `Cancel`.


--


#### Test Scenario H: Load Error Handling
1. Launch WPiOS and log into a dotcom account
2. Enable your phone's *airplane mode*
3. Tap over `Me` >  `Notifications`

As a result, you should get an alertView indicating that there was an error loading your settings. This alertView should allow you to `Retry` / `Cancel`.

--

Needs Review: @aerych (May i bother you with this one Eric?).

Thank you for Reading + Testing!
Reference: Issue #3807
